odoo/addons/base/res/ir_property.py

```diff
@@ -212,7 +212,9 @@ def clean(value):
             id = refs.pop(prop.res_id)
             value = clean(values[id])
             if value == default_value:
-                prop.unlink()
+                # avoid prop.unlink(), as it clears the record cache that can
+                # contain the value of other properties to set on record!
+                self._cr.execute("DELETE FROM ir_property WHERE id=%s", [prop.id])
```

Comment:
maybe add check access rule before
`prop.check_access_rights('unlink')`

Reply:
Indeed, and maybe `prop.check_access_rule('unlink')`, too :+1: